### DISCUSSION TITLE
v-data-table 2 visible items

### DISCUSSION THREAD

**Discussion by kellerza** [vuetifyjs/vuetify#17038] [Q&A]
How can I get the visible items programmatically?  [v-data-table v2](https://v2.vuetifyjs.com/en/api/v-data-table/)

I'm interested in the items in the correct order & after filters are applied

Thanks!

**Comment by simenkf** [comment]
What do you mean by getting visible items? If you want the items in the table to be sorted, you can click the header to sort by a specific column, or by multiple columns if you enable multi-sort.

**Reply by kellerza** [reply]
I need it programmatically or in other words, I want to access it in JavaScript.

Today I only have access to the input data, but after I've sorted & applied filters, I would like to copy the result to the clipboard, or export only the visible rows and keep the current order in which it is displayed.

**Reply by simenkf** [reply]
Then I would suggest you create a computed property applying a filter based on a search field. Create a v-textfield and place it in the header table slot. Bind a v-model to it, for instance titled searchString. Then you keep the original array with all original data, and use a computed property applying your search as the items value of the table. Something along the lines of:

`const tableItems = computed(() => originalData.filter(el => JSON.stringify(el).includes(searchString))`
